File: apps/web/src/lib/api/monitoring.ts
import { apiFetch } from "./client";

export interface MonitoringHealth {
  messages: {
    last_hour: number;
    last_24h: number;
    avg_latency_ms: number | null;
    p95_latency_ms: number | null;
  };
  intent_classifier: {
    classified_last_24h: number;
    sbert_only_pct: number | null;
    fallback_to_llm_pct: number | null;
  };
  tokens: {
    input_today: number;
    output_today: number;
  };
  conversations: {
    open: number;
    takeover: number;
    closed_today: number;
  };
  queue: {
    pending: number;
    workers_alive: number;
  };
}

export async function fetchMonitoringHealth(): Promise<MonitoringHealth> {
  return apiFetch<MonitoringHealth>("/api/v1/monitoring/health");
}

File: apps/api/src/chatbot_api/schemas/monitoring.py
"""Schemas for the internal monitoring endpoint (SW-28)."""

from pydantic import BaseModel


class MessagesMetrics(BaseModel):
    last_hour: int
    last_24h: int
    avg_latency_ms: float | None
    p95_latency_ms: float | None


class IntentClassifierMetrics(BaseModel):
    classified_last_24h: int
    sbert_only_pct: float | None
    fallback_to_llm_pct: float | None


class TokensMetrics(BaseModel):
    input_today: int
    output_today: int


class ConversationsMetrics(BaseModel):
    open: int
    takeover: int
    closed_today: int


class QueueMetrics(BaseModel):
    pending: int
    workers_alive: int


class MonitoringHealth(BaseModel):
    messages: MessagesMetrics
    intent_classifier: IntentClassifierMetrics
    tokens: TokensMetrics
    conversations: ConversationsMetrics
    queue: QueueMetrics
